### ISSUE TITLE
KDoc comments should not refer to private member variables

### ISSUE THREAD

**Issue by Whathecode** [detekt/detekt#4462]
I just ran into code which in KDoc comments pointed to a private member variable. E.g.

```
/**
  * Uses [answerCalculator] to find the answer to life, the universe, and everything.
  */
class Answer
{
    private val answerCalculator: String = "42"
    public fun getAnswer(): String = answerCalculator
}
```

## Expected Behavior of the rule
Given that `KDoc` is for _documentation_, intended to be useful to _callers_ of the thing being documented so that they don't need to read/understand encapsulated implementation details, wouldn't it make sense to consider it a code smell when documentation points to implementation internals?

## Context
This can help people to have some safeguards in place for writing better documentation. That said, I believe this is the first time I ran into code like this, so it may not be prevalent.

**Comment by cortinico** [comment]
> wouldn't it make sense to consider it a code smell when documentation points to implementation internals?

Potentially yes. However, I've been doing this multiple times in the past and I think it's acceptable. If we end up implementing this I believe it's going to be a rule disabled by default

**Comment by BraisGabin** [comment]
I agree that detekt should flag this.

**Comment by BraisGabin** [comment]
Related with #3657

**Comment by VitalyVPinchuk** [comment]
Hello,
may I try to work on the issue?

**Comment by cortinico** [comment]
Yup I've assigned it to you @[VitalyVPinchuk](https://github.com/VitalyVPinchuk)
